src/Icons/src/Registry/CacheIconRegistry.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Icons\Registry;
+
+use Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerInterface;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\UX\Icons\Exception\IconNotFoundException;
+use Symfony\UX\Icons\IconRegistryInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @internal
+ */
+final class CacheIconRegistry implements IconRegistryInterface, CacheWarmerInterface
+{
+    /**
+     * @param IconRegistryInterface[] $registries
+     */
+    public function __construct(private \Traversable $registries, private CacheInterface $cache)
+    {
+    }
+
+    public function get(string $name, bool $refresh = false): array
```

Comment:
Not sure of this "$refresh" ...  what would be the usage ?

Reply:
This forces the cache to be rebuilt (even if the icon is cached). See the warmUp method below for its usage.

Reply:
Is the cache cleared before ? If "yes", no need for this attribute... if "not"... well i think it should :) 

Let's see when we have sorted the "pool" question, we may have access to a "clearer" then that would remove any need for those things.

Reply:
Yes, this is true. I guess it's a a relic from when I had a configurable cache pool.

Reply:
I genuinely don't know if it's cleared or not, i see no "clear" in the base CacheInterface :/

Reply:
I've refactored this and removed the refresh.